app/src/main/java/unipd/se18/ocrcamera/CameraActivity.java

```diff
@@ -0,0 +1,123 @@
+ *  so we can support full range of Android devices (API 15 and higher is 100% demographic)
+ *  I used this interface for Camera (1) API to make the program
+ *      https://github.com/natario1/CameraView
+ *  working with master build of the interface
+ *  The documentation starts here
+ *      https://github.com/natario1/CameraView#usage
+ */
+
+package com.example.elia.compatibilitycamera;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v7.app.AppCompatActivity;
+import android.view.View;
+import android.content.pm.PackageManager;
+import android.support.annotation.NonNull;
+import android.widget.ImageButton;
+
+import com.otaliastudios.cameraview.CameraListener;
+import com.otaliastudios.cameraview.CameraView;
+import com.otaliastudios.cameraview.Gesture;
+import com.otaliastudios.cameraview.GestureAction;
+
+/**
+ * Activity for camera preview and taking photos
+ */
+public class CameraActivity extends AppCompatActivity {
+
+    private CameraView camera;
+    private boolean mCapturingPicture;
+
+    /**
+     * onCreate method of the Android Activity Lifecycle
+     * @param savedInstanceState the bundle of the last instance state saved
+     * @author Bedin Elia
+     */
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_camera);
+        camera = findViewById(R.id.viewCamera);
+
+        //interface method for automatic onResume, onPause, onDestroy
+        camera.setLifecycleOwner(this);
+
+        //Pinch to zoom support
+        camera.mapGesture(Gesture.PINCH, GestureAction.ZOOM);
+        //Tap to focus support
+        camera.mapGesture(Gesture.TAP, GestureAction.FOCUS_WITH_MARKER);
+        //Long tap to capture support (currently disabled, should be .CAPTURE)
+        camera.mapGesture(Gesture.LONG_TAP, GestureAction.NONE);
+
+        camera.addCameraListener(new CameraListener() {
+            /**
+             * Listining for photo taken by the camera
+             * @param jpeg the taken photo in array form
+             * @author Bedin Elia
+             */
+            public void onPictureTaken(byte[] jpeg) { onPicture(jpeg); }
+        });
+
+        ImageButton buttonPhoto = findViewById(R.id.capturePhoto);
+        buttonPhoto.setOnClickListener(new View.OnClickListener() {
+            /**
+             * Listening for button click, then call capturePhoto() to take a photo
+             * @param view the layout components
+             * @author Bedin Elia
+             */
+            @Override
+            public void onClick(View view) {
+                capturePhoto();
+            }
+        });
+
+    }
+
+    /**
+     * Method for taking a photo
+     * @author Bedin Elia
+     */
+    private void capturePhoto() {
+        if (mCapturingPicture) return;
+        mCapturingPicture = true;
+        camera.capturePicture();
```

Comment:
I'll try to explain you what I've understood (thanks in advance for the support):
This statement launches the callback for taking a picture and so the method onPictureTaken (line 62) is launched. Finally the onPictureTaken launches the onPicture method (line 95) that firstly sets the boolean variable mCapturingPicture to false. Is it right?
If mCapturingPicture is not already setted to false the capturePhoto method doesn't launch the camera.capturePicture() sentence but returns at the line 85. Is it right?

Reply:
You are correct, sorry if it was hard to understand.

Reply:
Don't worry, I've asked you for knowing if I've understand the functionality. So thanks for the feedback.